hack/bicep-types-radius/src/autorest.bicep/src/type-generator.ts

```diff
@@ -68,7 +68,7 @@ export function generateTypes(host: AutorestExtensionHost, definition: ProviderD
       const propertyDefinition = parseType(putProperty?.schema, getProperty?.schema);
       if (propertyDefinition) {
         const description = (putProperty?.schema ?? getProperty?.schema)?.language.default?.description;
-        const flags = parsePropertyFlags(putProperty, getProperty);
+        const flags = parsePropertyFlags(putProperty, getProperty, propertyName);
```

Comment:
So does this work because you're only passing in `propertyName` for top level properties? I'm assuming that there are other calls to `parsePropertyFlags` that don't pass in `propertyName`...

I wonder if it would be clearer to have a parameter called `isTopLevel` (or similar). 

This all feels like it should have a comment explaining **what** and **why**. It would be easy to someone else to work on this code and change the behavior by mistake.

Reply:
Yes, we only pass propertyName for top level properties. For other calls, we don't pass it in (i.e. https://github.com/radius-project/radius/blob/50394ac9c3ea2982419bbbf6e0255175759d221c/hack/bicep-types-radius/src/autorest.bicep/src/type-generator.ts#L470)

I'm not opposed to passing in another parameter to check if it's top level but since we'd need to pass in propertyName regardless to check if the field is 'location', I combined it into just one check for propertyName

Reply:
synced offline. A check for topLevel is already rolled into the check for propertyName so we'll just add a comment here explaining what's happening